service-node/backend/src/main/scala/org/enterprisedlt/fabric/service/node/process/DockerManagedBox.scala

```diff
@@ -409,6 +416,65 @@ class DockerManagedBox(
         }
     }
 
+
+    private def pullImageIfNeeded(image: String, forcePull: Boolean = false): Either[String, Unit] = {
+        if (forcePull) {
+            logger.info(s"Force pulling image $image")
+            pullImage(image)
+        } else {
+            findImage(image) match {
+                case Some(_) =>
+                    logger.info(s"Unable to find image $image locally")
+                    pullImage(image)
+
+                case None =>
+                    logger.debug(s"Image $image already exists")
+                    Right(())
+            }
+        }
+    }
+
+
+    private def findImage(image: String): Option[String] = {
+        docker
+          .listImagesCmd()
+          .exec()
+          .iterator()
+          .asScala
+          .toArray
+          .flatMap(image => Option(image.getRepoTags))
+          .flatten
+          .find(_ == image)
+    }
+
+
+    private def pullImage(image: String): Either[String, Unit] = {
+        try {
+            val pullImageCmd = image.split(":") match {
```

Comment:
pull this logic out i.e. change method to `pullImage(name: String, tag: String)`